### PR TITLE
AC Automated Test - LRAC-11381 and LRAC-11382 | master

### PR DESCRIPTION
**Pull request by ViniLopes20**
https://issues.liferay.com/browse/LRAC-11381 (Automation ticket 1)
https://issues.liferay.com/browse/LRAC-11382 (Automation ticket 2)
https://issues.liferay.com/browse/LRAC-10998 (Story)
[Testray Result](https://testray.liferay.com/home/-/testray/case_results?testrayBuildId=1882275728&testrayRunId=1882275779)

### CODE REVIEW

**Comment by ViniLopes20** [comment]
ci:test:sf

**Review by christianstokes — CHANGES_REQUESTED** [review]
This test looks really nice Vini. I think you can look into using Poshi's DateUtil.getTimeInMilliseconds for the date variables, but overall the test looks good to me.

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACUtils.macro [hunk 1]
```diff
@@ -451,6 +451,36 @@ definition {
...
+				locator1 = "ACUtils#INPUT_DATE",
+				value1 = "${year}");
+		}
+
+		var key_inputField = "rightMillis";
+		var millisecondsValue = selenium.getElementValue("ACUtils#INPUT_DATE");
+
+		return "${millisecondsValue}";
+	}
+
```
[the remark this answers is not in the document]
Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACUtils.macro:483]: Hey @christianstokes, that was a good suggestion. This macro works, but only for the current day, we cannot use it with a specific date. So I think the best solution would be to use this site to do that.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/IdentityEvents.testcase [hunk 2]
```diff
@@ -199,6 +200,224 @@ definition {
...
+					var month = ${currentMonth} - 1;
+				}
+			}
+			else {
+				var day = DateUtil.getDate("-8");
+				var month = "${currentMonth}";
+				var year = "${currentYear}";
+			}
+		}
+
```
[the remark this answers is not in the document]
Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/IdentityEvents.testcase:354]: Hey @christianstokes, I didn't understand the logic you wanted to show for this situation. Do you think that if we decrease the day in milliseconds with 604,800,000 we should get the result of 7 days ago?

[the remark this answers is not in the document]
Comment (christianstokes) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/IdentityEvents.testcase:354]: Clarified in a conversation with Vini over slack. He will try using Poshi's dateUtil to get the current date in millis, then subtracting 86,400,000 per day to get to the value needed for the test.

**Comment by christianstokes** [comment]
Hey @ViniLopes20 any updates on this? Since https://issues.liferay.com/browse/POSHI-374 is done?

**Comment by ViniLopes20** [comment]
> Hey @ViniLopes20 any updates on this? Since https://issues.liferay.com/browse/POSHI-374 is done?

Hey @christianstokes, I talked to Calum and he said he finished the implementation, but the change won't be available until the next Poshi release. I'm following the channel to see when it's released.

**Comment by ViniLopes20** [comment]
ci:test:sf

**Comment by ViniLopes20** [comment]
ci:test:sf

**Comment by ViniLopes20** [comment]
ci:test:sf

**Comment by ViniLopes20** [comment]
Hey @christianstokes, the Poshi release is done and I ran both tests locally and they are passing after this ticket. I thought of using the function we asked for Calum _getTimeInMilliseconds_, but in this case, we would need to create a date logic to pass the date 7 days or 1 day ago to the function, with the month and year changes too, it would be very complex.

I think using _MathUtil.difference_ will make it clearer for testing. When I ran SF it changed _MathUtil.difference_ to use "-", I had [asked](https://liferay.slack.com/archives/CD7939WBE/p1656542200082959?thread_ts=1656534095.605969&cid=CD7939WBE) this question to Kenji a few months ago and he said it was expected.

**Review by christianstokes — APPROVED** [review]
Looks great @ViniLopes20. This is a really cool test, and I'm glad everything worked out with the Poshi team. It's always fun to use a totally new function 😀

**Comment by christianstokes** [comment]
ci:forward

**Pull request closed without merging**